nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -931,7 +933,7 @@ def get_batch_on_this_context_parallel_rank(self, batch):
                     val = val.view(*val.shape[0:seq_dim], -1, *val.shape[(seq_dim + 2) :])
                     batch[key] = val
 
-        batch['num_valid_tokens_in_ub'] = num_valid_tokens_in_ub
+        batch['num_valid_tokens_per_sample'] = num_valid_tokens_per_sample
```

Comment:
so `num_valid_tokens_per_sample` can be none if batch.get('loss_mask') is None? And I didn't see None handling in the loss calculation.

Reply:
It shouldn't be None in practice -- I think `batch["loss_mask"]` can only be None on PP ranks that don't compute the loss. I didn't look at this very closely because I am not changing this specific aspect in this PR (it was the same before).

Reply:
okay. it makes sense.  The none can happen for intermediate PP ranks.